docs/DeveloperGuide.md

```diff
@@ -351,178 +363,278 @@ Use case ends.
 
-1. Avenger requests to mark a task as done.
-2. JARVIS displays the list of tasks with the aforementioned task marked as done.
+**Use case: UC16 - Mark a student as present for a lesson**
+
+Preconditions: There are existing students and lessons in JARVIS.
+
+**MSS**
+1. Avenger performs <ins>view lessons(UC3)</ins>.
+2. Avenger requests to mark a student present in a lesson.
+3. JARVIS marks the student present in the lesson.
 
 Use case ends.
 
 **Extensions:**
 
-* 1a. JARVIS fails to understand request.
-  * 1a1. JARVIS tells Avenger to make a request again.
+- 2a. JARVIS fails to understand request.
+  - 2a1. JARVIS tells Avenger to make a request again.
+  - 
+    Use case resumes from step 2.
+  
+- 2b. Specified lesson not found in JARVIS.
+  - 2b1. JARVIS informs Avenger that the lesson does not exist.
+  - 2b2. JARVIS tells Avenger to make a request again.
+  
+    Use case resumes from step 2.
+  
+- 2c. Specified student not found in JARVIS.
+  - 2c1. JARVIS informs Avenger that the student does not exist in the lesson.
+  - 2c2. JARVIS tells Avenger to make a request again.
+  
+    Use case resumes from step 2.
+  
+**Use case: UC17 - Mark a student as absent for a lesson**
 
-    Use case resumes from step 1.
+Refer to <ins>UC16 - Mark a student as present for a lesson</ins> with the only difference being mark a student as absent instead of present.
 
-* 1b. JARVIS detects invalid task.
-  * 1b1. JARVIS informs Avenger that the task does not exist.
-  * 1b2. JARVIS displays the list of tasks.
-  * 1b3. JARVIS tells Avenger to make a request again.
+**Use case: UC18 - Add note for a student in a lesson**
 
-    Use case resumes from step 1.
+Preconditions: There are existing students and lessons in JARVIS.
 
-**Use case: UC9 - Mark task as not done**
+1. Avenger performs <ins>view lessons(UC3)</ins>.
+2. Avenger requests to add a note for a student in a lesson.
+3. JARVIS adds the note for the student to the lesson.
 
-Preconditions: There are existing tasks in JARVIS
+Use case ends.
+
+**Extensions:**
+- 2a. JARVIS fails to understand request.
+    - 2a1. JARVIS tells Avenger to make a request again.
+  
+      Use case resumes from step 2.
+  
+- 2b. Specified lesson not found in JARVIS.
+    - 2b1. JARVIS informs Avenger that the lesson does not exist.
+    - 2b2. JARVIS tells Avenger to make a request again.
+  
+      Use case resumes from step 2.
 
-Guarantees: Specified task is marked as not done
+- 2c. Specified student not found in JARVIS.
+    - 2c1. JARVIS informs Avenger that the student does not exist in the lesson.
+    - 2c2. JARVIS tells Avenger to make a request again.
 
-**MSS:**
+      Use case resumes from step 2.
+
+
+**Use case: UC19 - Delete note for a student in a lesson**
+
+Refer to <ins>UC18 - Add note for a student in a lesson</ins> with the only difference being delete note instead of add note.
```

Comment:
I think this use case might be slightly different because there is the possibility of the note index being invalid